app/V1Module/presenters/SubmitPresenter.php

```diff
@@ -101,6 +107,20 @@ class SubmitPresenter extends BasePresenter {
 
 
   /**
+   * Determine if given user can submit solutions to assignment.
+   * @param Assignment $assignment
+   * @param User|NULL $user
+   * @return bool
+   */
+  public function canReceiveSubmissions(Assignment $assignment, User $user = null) {
```

Comment:
This was moved from the Assignment entity, right? This is certainly a better place, but I would make this method private.

Reply:
Definitely a good idea :-)